### ISSUE TITLE
mochify --watch does not recover from syntax errors

### ISSUE THREAD

**Issue by Widdershin** [mantoni/mochify.js#123]
When running `mochify --watch`, if you change a file so that it is not syntactically valid, it will correctly report the SyntaxError, but then when the error is rectified `mochify` does not rerun the tests.

Steps to reproduce:
Create a test file, e.g:
```js
/* globals describe, it */

var assert = require('assert');

describe('mocha --watch', function () {
  it('does not handle syntax errors', function () {
    assert(true);
  });
});
```
Run `mochify --watch`.
Change the test file so that it has a syntax error (e.g deleting the brace off the end of a function declaration).
`mochify` will then report an error, like so:

```
Error: Parsing file /Users/nickj/Projects/cycle-restart/test/syntax-error-test.js: Unexpected token (7:4)

Error: Exit 1
```

Then change the file so that it is again syntactically valid.

Expected behaviour:
`mochify` should rerun the test.

Actual behaviour:
`mochify` does nothing, and needs to be restarted before it will run the test again.

It's worth noting that `mocha --watch` gracefully handles syntax errors. It would be nice if `mochify` also did so.

**Comment by mantoni** [comment]
Thanks for reporting. I've seen this myself too. I'd appreciate anyones help fixing this.

**Comment by mantoni** [comment]
Oh, this was actually surprisingly easy to fix. Released a patch in `2.14.3`. Please update and test again.

**Comment by Widdershin** [comment]
@mantoni the fix totally works, thank you tons for such an amazing turnaround time.
